docs/src/main/tut/schema-evolution/avro.md

```diff
@@ -0,0 +1,292 @@
+
+No action required. However, keep in mind that the value will be ignored when old clients include it in the request.
+
+* Before:
+
+```scala
+case class Request(a: String, b: Int)
+```
+
+* After:
+
+```scala
+case class NewRequest(a: String)
+```
+
+### Modifying the Response (Server side)
+
+#### I: Adding a new non-optional field
+
+In this case, the old clients will ignore the value of the new field, so everything will be safe in terms of backward compatibility.
+
+* Before:
+
+```scala
+case class Response(a: String, b: Int)
+```
+
+* After:
+
+```scala
+case class NewResponse(a: String, b: Int, c: Boolean)
+```
+
+#### J: Adding a new optional field
+
+This would be just a particular case of the previous scenario, where the default value would be `None`, in the case of `Scala`.
+
+#### K: Adding a new item to a coproduct
+
+In this scenario, the old clients will fail when the result is including the new item. Hence, the solution would be to provide a default value to the old coproduct and creating a new field with the new coproduct.
+
+* Before:
+
+```scala
+case class Response(a: Int :+: String :+: CNil)
+```
+
+* After:
+
+```scala
+case class NewResponse(
+      a: Int :+: String :+: CNil = Coproduct[Int :+: String :+: CNil](0),
+      b: Int :+: String :+: Boolean :+: CNil)
+```
+
+#### L: Removing item from a coproduct
+
+No action will be required in this case.
+
+* Before:
+
+```scala
+case class Response(a: Int :+: String :+: CNil)
+```
+
+* After:
+
+```scala
+case class NewResponse(a: Int :+: CNil)
+```
+
+#### M: Replacing item from a coproduct
+
+As long as the value of the coproduct belongs to the previous version, the old client should be able to accept the response as valid. Thus, we would need to follow the same approach as above when _Adding a new item to a coproduct_.
+
+* Before:
+
+```scala
+case class Response(a: Int :+: String :+: CNil)
+```
+
+* After:
+
+```scala
+case class NewResponse(a: Int :+: Boolean :+: CNil)
+```
+
+#### N: Changing the type of an existing field
+
+It will require providing a default value for the previous type, and then, we would need to create a new field with the new type.
+
+* Before:
+
+```scala
+case class Response(a: String, b: Int)
+```
+
+* After:
+
+```scala
+case class NewResponse(a: String, b: Int = 123, c: Boolean)
+```
+
+#### O: Renaming a field
+
+It will require to keep the previous name and type with a default value, and then, adding a new field with the same type.
+
+* Before:
+
+```scala
+case class Response(a: String, b: Int)
+```
+
+* After:
+
+```scala
+case class NewResponse(a: String, b: Int = 123, c: Int)
+```
+
+#### P: Removing a field
+
+This evolution should never happen since we would lose backward compatibility. Nonetheless, it would work only under the special case that the old response has a default value for the field that we want to delete, where this operation would be feasible by removing the field in the new version of the server response.
```

Comment:
I understood this occurs because the client will use its default value, isn't it?